code/datums/quirks/_quirk_static_data.dm

```diff
@@ -0,0 +1,74 @@
+GLOBAL_LIST_INIT_TYPED(all_quirk_static_data, /datum/quirk_static_data, generate_quirk_static_data())
+
+/// Constructs [GLOB.all_quirk_static_data] by iterating through a typecache of pregen data, ignoring abstract types, and instantiating the rest.
+/proc/generate_quirk_static_data()
+	RETURN_TYPE(/list/datum/quirk_static_data)
+
+	var/list/datum/quirk_static_data/all_pregen_data = list()
+
+	for (var/datum/quirk_static_data/iterated_path as anything in typecacheof(path = /datum/quirk_static_data, ignore_root_path = TRUE))
+		if (initial(iterated_path.abstract_type) == iterated_path)
+			continue
+
+		if (!isnull(all_pregen_data[initial(iterated_path.associated_typepath)]))
+			stack_trace("pre-existing pregen data for [initial(iterated_path.associated_typepath)] when [iterated_path] was being considered: [all_pregen_data[initial(iterated_path.associated_typepath)]]. \
+						this is definitely a bug, and is probably because one of the two pregen data have the wrong quirk typepath defined. [iterated_path] will not be instantiated")
```

Comment:
Why is this indented so far? Only ever indent up by one.

Reply:
im not sure why github makes it look so heavily indented, but heres what it looks like 
![image](https://github.com/tgstation/tgstation/assets/59709059/0a83a2f8-6697-4201-98bd-6c8d51c5bc6b)

Reply:
sure